app/src/ui/app-theme.tsx

```diff
@@ -44,19 +58,128 @@ export class AppTheme extends React.PureComponent<IAppThemeProps> {
     const newThemeClassName = `theme-${getThemeName(themeToDisplay)}`
     const body = document.body
 
-    if (body.classList.contains(newThemeClassName)) {
-      return
+    if (
+      !body.classList.contains(newThemeClassName) ||
+      (body.classList.contains('theme-custom') && !this.props.useCustomTheme)
+    ) {
+      this.clearThemes()
+      body.classList.add(newThemeClassName)
     }
+  }
 
-    this.clearThemes()
+  /**
+   * This takes a custom theme object and applies it over top either our dark or
+   * light theme dynamically creating a new variables style sheet.
+   *
+   * It uses the background color of the custom theme to determine if the custom
+   * theme should be based on the light or dark theme. This is most important
+   * for the diff syntax highlighting.
+   *
+   * Currently, our only custom theme is a high-contrast theme, thus there are
+   * styles that are specifically added for this purpose such as adding borders
+   * or backgrounds to things that didn't have borders in our non-high-contrast
+   * themes.
+   *
+   * @param customTheme
+   */
+  private setCustomTheme(customTheme: ICustomTheme) {
+    const { background, text, activeItem, activeText, border } = customTheme
+    const body = document.body
+
+    if (!body.classList.contains('theme-custom')) {
+      body.classList.add('theme-custom')
+      // This is important so that code diff syntax colors are legible if the
+      // user customizes to a light vs dark background. Tho, the code diff does
+      // still use the customizable text color for some of the syntax text so
+      // user can still make things illegible by choosing poorly.
+      const themeBase = isHexColorLight(background)
+        ? 'theme-light'
+        : 'theme-dark'
+      body.classList.add(themeBase)
+    }
+
+    const styles = document.createElement('style')
+    styles.setAttribute('type', 'text/css')
+
+    const secondaryActiveColor = lightenDarkenHexColor(activeItem, 20)
+    const secondaryBackgroundColor = lightenDarkenHexColor(background, 20)
+
+    const highContrastSpecific = `
+        --box-selected-active-border: 2px solid ${border};
+        --list-item-hover-border: 2px solid ${border};
+
+        --secondary-button-hover-border-width: 2px;
+
+        --tab-bar-box-shadow: none;
+
+        --diff-add-border: 1px solid green;
+        --diff-delete-border: 1px solid crimson;
+        --tab-bar-hover-border: 2px solid ${border} !important;
+        --tab-bar-item-border: 2px solid ${background};
+        --foldout-border: 1px solid ${border};
+        --horizontal-bar-active-color: ${activeItem};
+        --horizontal-bar-active-text-color: ${activeText};
+    `
+
+    styles.appendChild(
+      document.createTextNode(
```

Comment:
To note... This feels a bit hacky to me because you lose the context of the css syntax. However, from what research I did, it seemed like about the only way to go about programmatically changing theme variables from js. 

In fact.. now that I think about it, it may be nice to pull the meat of this style generating code out into a library function that outputs the stylesheet string so we can run "lint type checking" unit tests against it.